app/main/views/suppliers.py

```diff
@@ -147,9 +148,74 @@ def import_supplier(supplier_id):
 
         supplier.contact_information.append(contact_information)
 
-    db.session.add(supplier)
+        db.session.add(supplier)
+
+    try:
+        db.session.commit()
+    except IntegrityError as e:
+        db.session.rollback()
+        abort(400, "Database Error: {0}".format(e))
+
+    return jsonify(suppliers=supplier.serialize()), 201
+
+
+@main.route('/suppliers', methods=['POST'])
+def create_supplier():
+    supplier_data = get_json_from_request()
+
+    json_has_required_keys(
+        supplier_data,
+        ['suppliers']
+    )
+
+    supplier_data = supplier_data['suppliers']
+    supplier_data = drop_foreign_fields(supplier_data, ['links'])
+
+    json_has_required_keys(
+        supplier_data,
+        ['name', 'contactInformation']
+    )
+
+    contact_informations_data = [
+        drop_foreign_fields(contact_data, ['links'])
+        for contact_data in supplier_data['contactInformation']
+        ]
+
+    supplier_data['contactInformation'] = contact_informations_data
+
+    for contact_information_data in contact_informations_data:
+        json_has_required_keys(
+            contact_information_data,
+            ['contactName', 'email', 'postcode']
+        )
+
+    validate_new_supplier_json_or_400(supplier_data)
+
+    supplier_data = drop_foreign_fields(
+        supplier_data,
+        ['contactInformation']
+    )
+
+    supplier = Supplier()
+    supplier.update_from_json(supplier_data)
+
+    for contact_information_data in contact_informations_data:
```

Comment:
Can we split this to a separate "create contact information" endpoint? This is how updates work right now, so it should make it easier to reuse the frontend code/forms.